TESTS/mbedmicro-rtos-mbed/timer/main.cpp

```diff
@@ -0,0 +1,872 @@
+    /* Start the timer. */
+    p_timer->start();
+
+    /* -- Simulate that current time is equal to 1 ms -- */
+    curr_ticker_us_val = 1000;
+
+    /* Stop the timer. */
+    p_timer->stop();
+
+    /* Read timer value using each read function. */
+    read_val = p_timer->read();
+    read_ms_val = p_timer->read_ms();
+    read_us_val = p_timer->read_us();
+    read_hr_us_val = p_timer->read_high_resolution_us();
+
+    /* Check results - 1 ms has elapsed. */
+    TEST_ASSERT_EQUAL_FLOAT(0.001000f, read_val);
+    TEST_ASSERT_EQUAL(1, read_ms_val);
+    TEST_ASSERT_EQUAL(1000, read_us_val);
+    TEST_ASSERT_EQUAL(1000, read_hr_us_val);
+
+    /* Start the timer. */
+    p_timer->start();
+
+    /* -- Simulate that current time is equal to 125 ms -- */
+    curr_ticker_us_val = 125000;
+
+    /* Stop the timer. */
+    p_timer->stop();
+
+    /* Read timer value using each read function. */
+    read_val = p_timer->read();
+    read_ms_val = p_timer->read_ms();
+    read_us_val = p_timer->read_us();
+    read_hr_us_val = p_timer->read_high_resolution_us();
+
+    /* Check results - 125 ms have elapsed. */
+    TEST_ASSERT_EQUAL_FLOAT(0.125000f, read_val);
+    TEST_ASSERT_EQUAL(125, read_ms_val);
+    TEST_ASSERT_EQUAL(125000, read_us_val);
+    TEST_ASSERT_EQUAL(125000, read_hr_us_val);
+
+    /* Start the timer. */
+    p_timer->start();
+
+    /* -- Simulate that current time is equal to 1 s -- */
+    curr_ticker_us_val = 1000000;
+
+    /* Stop the timer. */
+    p_timer->stop();
+
+    /* Read timer value using each read function. */
+    read_val = p_timer->read();
+    read_ms_val = p_timer->read_ms();
+    read_us_val = p_timer->read_us();
+    read_hr_us_val = p_timer->read_high_resolution_us();
+
+    /* Check results - 1 s has elapsed. */
+    TEST_ASSERT_EQUAL_FLOAT(1.000000f, read_val);
+    TEST_ASSERT_EQUAL(1000, read_ms_val);
+    TEST_ASSERT_EQUAL(1000000, read_us_val);
+    TEST_ASSERT_EQUAL(1000000, read_hr_us_val);
+
+    /* Start the timer. */
+    p_timer->start();
+
+    /* -- Simulate that current time is equal to 125 s -- */
+    curr_ticker_us_val = 125000000;
+
+    /* Stop the timer. */
+    p_timer->stop();
+
+    /* Read timer value using each read function. */
+    read_val = p_timer->read();
+    read_ms_val = p_timer->read_ms();
+    read_us_val = p_timer->read_us();
+    read_hr_us_val = p_timer->read_high_resolution_us();
+
+    /* Check results - 125 s have elapsed. */
+    TEST_ASSERT_EQUAL_FLOAT(125.000000f, read_val);
+    TEST_ASSERT_EQUAL(125000, read_ms_val);
+    TEST_ASSERT_EQUAL(125000000, read_us_val);
+    TEST_ASSERT_EQUAL(125000000, read_hr_us_val);
+
+    /* Start the timer. */
+    p_timer->start();
+
+    /* -- Simulate that current time is equal to MAX_INT_32 us -- */
+
+    /* Note that timers are based on 32-bit int microsecond counters,
+     * so can only time up to a maximum of 2^31-1 microseconds i.e.
+     * 2147483647 us (about 35 minutes). */
+    curr_ticker_us_val = 2147483647;
```

Comment:
curr_ticker_us_val is of unsigned integer type. In case when the value would be greater than MAX_INT, then we have inconsistency. In such case read functions would return negative values.

I have assumed that the max tested measured time can not exceed MAX_INT based on information from mBed handbook (see Warning):
https://developer.mbed.org/handbook/Timer
https://docs.mbed.com/docs/mbed-os-api-reference/en/latest/APIs/tasks/Timer/

According to the documentation Timer is dedicated to measure small times (between microseconds and seconds).